src/connections/connections.controller.ts

```diff
@@ -0,0 +1,97 @@
+import { Controller, Delete, Get, Param, Post } from '@nestjs/common';
+import { CurrentUser } from 'src/decorators/current-user.decorator';
+import { ConnectionsService } from './connections.service';
+import { ConnectionDto } from './DTO/Connection.dto';
+import {
+  ApiBadRequestResponse,
+  ApiBearerAuth,
+  ApiNotFoundResponse,
+} from '@nestjs/swagger';
+import { User } from '@prisma/client';
+import { Public } from 'src/decorators/public.decorator';
+
+@ApiBearerAuth()
+@Controller('connections')
+export class ConnectionsController {
+  constructor(private connectionsService: ConnectionsService) {}
+  /**
+   *
+   * Get all of the current user's connections
+   */
+  @Get()
+  async getAllConnections(@CurrentUser() user: User): Promise<ConnectionDto[]> {
+    return this.connectionsService.getUserConnections(user.id);
+  }
+
+  /**
+   * Get "suggested for review" connections.
+   */
+  @Get('/suggested')
+  async getSuggestedConnections(
+    @CurrentUser() user: User,
+  ): Promise<ConnectionDto[]> {
+    return this.connectionsService.getUserConnections(user.id);
+  }
+
+  /**
+   * Search an user by query.
+   */
+  @Get('/search/:query')
+  @Public()
+  @ApiBadRequestResponse()
+  async searchUser(
+    @Param('query') query: string,
+    @CurrentUser() user?: User,
+  ): Promise<ConnectionDto[]> {
+    return this.connectionsService.searchUsers(user?.id, query);
+  }
+
+  /**
+   * Create a connection between current User and another user.
+   */
+  @Post('/connect/:userId')
```

Comment:
I think a better name for this would be `add`. Similarly, for the delete endpoint it can be `remove`. This will look pretty meaningful when put together:
- `GET /api/connection/add/:userId`
- `DELETE /api/connection/remove/:userId`

Reply:
The way it is right now it is in alignment with RESTful standards. The verb "POST" signifies a resource creation, and DELETE signifies a resource removal.
Adding verbs like add and remove in the endpoint path can be redundant and not as clean as using HTTP methods to indicate the action. RESTful best practices suggest using nouns in URLs and verbs in HTTP methods.

https://mglaman.dev/blog/post-or-put-patch-and-delete-urls-are-cheap-api-design-matters
https://www.mscharhag.com/api-design/rest-deleting-resources

Reply:
True, I know it is bit of a gamble in here. The thing is, feels that the `connect` word might specify an action and not a resource. Truly we can get rid of `add` and `delete` since the methods will signify that. I think we can go with `/api/connection/:userId` as the path and `DELETE` and `POST` as the methods?